slowfast/models/resnet_helper.py

```diff
@@ -107,6 +115,288 @@ def forward(self, x):
         return x
 
 
+class TimeFreqBottleneckTransform_v1(nn.Module):
+    """
+    The transformation function that decouples time 
+    and frequency axis in log-mel-spectrogram inputs, 
+    as described in the AVSlowFast paper.  
+    1x1x3, 1x3x1, 1x1x1
+    """
+
+    def __init__(
+        self,
+        dim_in,
+        dim_out,
+        temp_kernel_size,
+        stride,
+        dim_inner,
+        num_groups,
+        stride_1x1=False,
+        inplace_relu=True,
+        eps=1e-5,
+        bn_mmt=0.1,
+        dilation=1,
+        norm_module=nn.BatchNorm3d,
+    ):
+        """
+        Args:
+            dim_in (int): the channel dimensions of the input.
+            dim_out (int): the channel dimension of the output.
+            temp_kernel_size (int): the temporal kernel sizes of the middle
+                convolution in the bottleneck.
+            stride (int): the stride of the bottleneck.
+            dim_inner (int): the inner dimension of the block.
+            num_groups (int): number of groups for the convolution. num_groups=1
+                is for standard ResNet like networks, and num_groups>1 is for
+                ResNeXt like networks.
+            stride_1x1 (bool): if True, apply stride to 1x1 conv, otherwise
+                apply stride to the 3x3 conv.
+            inplace_relu (bool): if True, calculate the relu on the original
+                input without allocating new memory.
+            eps (float): epsilon for batch norm.
+            bn_mmt (float): momentum for batch norm. Noted that BN momentum in
+                PyTorch = 1 - BN momentum in Caffe2.
+            dilation (int): size of dilation.
+            norm_module (nn.Module): nn.Module for the normalization layer. The
+                default is nn.BatchNorm3d.
+        """
+        super(TimeFreqBottleneckTransform_v1, self).__init__()
+        self.temp_kernel_size = temp_kernel_size
+        self._inplace_relu = inplace_relu
+        self._eps = eps
+        self._bn_mmt = bn_mmt
+        self._stride_1x1 = stride_1x1
+        self._construct(
+            dim_in, 
+            dim_out, 
+            stride, 
+            dim_inner, 
+            num_groups,
+            dilation,
+            norm_module,
+        )
+
+
+    def _construct(self, dim_in, dim_out, stride, dim_inner, num_groups, 
+                   dilation, norm_module):
+        (str1x1, str3x3) = (stride, 1) if self._stride_1x1 else (1, stride)
+        # 1x3x1, BN, ReLU.
+        self.t = nn.Conv3d(
+            dim_in,
+            dim_inner,
+            [1, 3, 1],
+            stride=[1, str3x3, str3x3],
+            padding=[0, 1, 0],
+            groups=num_groups,
+            bias=False,
+        )
+        self.t_bn = norm_module(
+            dim_inner, eps=self._eps, momentum=self._bn_mmt
+        )
+        self.t_relu = nn.ReLU(inplace=self._inplace_relu)
+        
+        # 1x1x3, BN, ReLU.
+        self.f = nn.Conv3d(
```

Comment:
Hi, I am trying to re-implement AVSlowfast, and feel confused here.

In the paper (Table 1), the dim_inner should be the same as the planes, but the code here shows that the dim_inner is 2 * planes. i.e., for res2: [3×1, 1×3], 32 should be [3×1, 1×3], 64.